javascript/src/test/components/BasicCourseSearch/BasicCourseTable.test.js

```diff
@@ -129,7 +130,34 @@ describe("BasicCourseTable tests", () => {
   	const {getByText} = render(<BasicCourseTable classes = {courseFixtures.classesLectureAndSections} />);
   	expect( getBackgroundColor(getByText, "0101") ).toBe("#EDF3FE");
   });
-  
+
+  test("add buttons tester for sections with add buttons", ()=> {  
+    const {getByTestId} = render(<BasicCourseTable classes = {courseFixtures.classesLectureAndSections} />);
+    expect(getByTestId('add-button-07500')).toBeInTheDocument();
+    const addButton = getByTestId('add-button-07500');
+    userEvent.click(addButton);
+  });
+  test("add buttons for lecture with sections", ()=> {
+    const {getByTestId} = render(<BasicCourseTable classes = {courseFixtures.classesLectureAndSections} />);
+    try {
+        expect(getByTestId('add-button-07492')).not.toBeVisible();
+    } catch (error) {
+        expect(true).toBeTruthy();
```

Comment:
Agreed with Jayleen

Reply:
Hi, 

I added it because whenever I leave line 143 I get a TestingLibraryElementError because it is unable to find an element with a test id 'add-button-07492'. I tried the same thing with not.toBeInDocument() and that also threw an error. Please let me know if you have any recommendations.

Reply:
There is an error thrown with getByTestId('add-button-07492') because there is no button with test id 'add-button-07492'. 

Try expect(queryByTestId('add-button-07492')).not.toBeInTheDocument();

queryByTestId checks if an element exists or not. If it doesn't exist it won't throw an error as with the problem with using getByTestId. It will instead just return null.